test/tbb/test_concurrent_lru_cache.cpp

```diff
@@ -125,3 +126,25 @@ TEST_CASE("basic test for eviction of only unused items 2") {
     REQUIRE_MESSAGE(is_correct, "cache should not evict items in use");
 }
 
+//! \brief \ref error_guessing
+TEST_CASE("basic test for handling case when number_of_lru_history_items is zero") {
+    auto counter_func = [] (int) {
+        return utils::LifeTrackableObject{};
+    };
+    using cache_type =  tbb::concurrent_lru_cache<int, utils::LifeTrackableObject, decltype(counter_func)>;
+    cache_type cache{counter_func, 0};
+    
+    for(int i = 0; i < 10; ++i) {
+        const utils::LifeTrackableObject* obj_addr = &cache[1].value();
+        REQUIRE_MESSAGE(utils::LifeTrackableObject::is_alive(obj_addr) == false, "when number_of_lru_history_items is zero, element must be erased after use");
+    }
+
+    cache_type::handle h = cache[1];
+    const utils::LifeTrackableObject* obj_addr = &h.value();
+    auto& object_set = utils::LifeTrackableObject::set();
+    for(int i = 0; i < 10; ++i) {
+        cache[1];
+        REQUIRE_MESSAGE(utils::LifeTrackableObject::is_alive(obj_addr), "there is still a reference for key=1");
```

Comment:
It seems like the message does not report what is going wrong. Consider something like: "The object with the key=1 was destroyed but should not"